packages/ses/src/make-eval-function.js

```diff
@@ -19,7 +23,15 @@ export const makeEvalFunction = safeEvaluate => {
         // rule. Track.
         return source;
       }
-      return safeEvaluate(source);
+      if (legacyHermesTaming === 'unsafe') {
+        throw TypeError(
+          `Legacy Hermes unsupported eval() called with string arguments cannot be tamed safe under legacyHermesTaming ${legacyHermesTaming}
+  See: https://github.com/facebook/hermes/issues/1056
+  See: https://github.com/endojs/endo/discussions/1944
+Did you mean evalTaming: 'unsafeEval'?`,
+        );
+      }
+      return evaluator(source);
```

Comment:
this is better than `Uncaught SyntaxError: 2:5:invalid statement encountered`

Reply:
```suggestion
      try {
        evaluator(source);
      } catch (e) {
        if (
          e.name === 'SyntaxError' &&
          e.message.includes('invalid statement encountered')
        ) {
          throw TypeError(
            `Legacy Hermes unsupported eval() with strings arguments cannot be tamed safe under legacyHermesTaming ${legacyHermesTaming}
  See: https://github.com/facebook/hermes/issues/1056
  See: https://github.com/endojs/endo/discussions/1944
Did you mean evalTaming: 'unsafeEval'?`,
          );
        }
      }
```

Reply:
try/catch preferred to support safeEval on hosts supporting `with` statement

Reply:
else re throw original error, in case yet-another-es-engine fails on evaluateFactory